### DISCUSSION TITLE
how to set  workspaceFolder to null in initialing Json RPC message

### DISCUSSION THREAD

**Discussion by cjayashantha** [TypeFox/monaco-languageclient#651] [Q&A] [answered]
I am trying to connect with the xtext service. below is the initial message. I removed the capabilities for the readability of the message.

this message gives an error because of the **workspaceFolders**. if it is set to null, this works correctly.

```json
{
    "jsonrpc": "2.0",
    "id": 0,
    "method": "initialize",
    "params": {
        "processId": null,
        "clientInfo": {
            "name": "Code - OSS",
            "version": "1.88.1"
        },
        "locale": "en",
        "rootPath": null,
        "rootUri": null,
        "capabilities": {},
        "trace": "off",
        "workspaceFolders": [
            {
                "uri": "inmemory://model/",
                "name": ""
            }
        ]
    }

}
```

how to set workspace folder  to null ?

for the code, please refer to this.
[https://github.com/TypeFox/monaco-languageclient/issues/649](https://github.com/TypeFox/monaco-languageclient/issues/649)

Thanks in advance

**Comment by kaisalmen** [comment, marked as answer]
@cjayashantha there workspace folder should not be an in-memory thing. The language server needs to be able access this folder in its local file-system usually (I am saying usually here as the various language servers out there may handle this differently).
